OpenSim/Simulation/Test/testFrames.cpp

```diff
@@ -468,70 +461,52 @@ void testStationOnFrame()
 void testStationInGround()
 {
     SimTK::Vec3 tolerance(SimTK::Eps);
-    
+    SimTK::MultibodySystem system;
+    SimTK::SimbodyMatterSubsystem matter(system);
+    SimTK::GeneralForceSubsystem forces(system);
+
     cout << "Running testStationInGround" << endl;
     
     Model* pendulum = new Model("double_pendulum.osim");
-    // Get "rod1" frame
-    const OpenSim::Body& rod1 = pendulum->getBodySet().get("rod1");
-    const SimTK::Vec3& com = rod1.get_mass_center();
-    // Create station aligned with rod1 com in rod1_frame
+    const OpenSim::Body& rod2 = pendulum->getBodySet().get("rod2");
+    
+    // Define and add a frame to the rod2 body
+    SimTK::Transform X_RO;
+    X_RO.setP(SimTK::Vec3(0.2, -0.2, 0));
+    X_RO.updR().setRotationFromAngleAboutAxis(SimTK::Pi/2, SimTK::ZAxis);
+    PhysicalOffsetFrame* offsetFrame = new PhysicalOffsetFrame(rod2, X_RO);
+    offsetFrame->setName("myExtraFrame");
+    pendulum->addFrame(offsetFrame);
+    
+    // Create station in the extra frame
     Station* myStation = new Station();
-    myStation->set_location(com);
-    myStation->updConnector<PhysicalFrame>("reference_frame")
-    .setConnecteeName("rod1");
+    const SimTK::Vec3 point(0.5, 1, -1.5);
+    myStation->set_location(point);
+    myStation->updConnector<PhysicalFrame>("reference_frame").setConnecteeName("myExtraFrame");
     pendulum->addModelComponent(myStation);
-    // myStation should coincide with com location of rod1 in ground
-    SimTK::State& s = pendulum->initSystem();
-    for (double ang = 0; ang <= 90.0; ang += 10.){
-        double radAngle = SimTK::convertDegreesToRadians(ang);
-        const Coordinate& coord = pendulum->getCoordinateSet().get("q1");
-        coord.setValue(s, radAngle);
-        
-        // Get the station in ground via Simbody
-        SimTK::Vec3 comBySimbody = rod1.getMobilizedBody().findStationLocationInGround(s, com);
-        
-        // test findLocationInGround
-        SimTK::Vec3 stationInGround = myStation->findLocationInGround(s);
-        
-        SimTK_TEST_EQ(stationInGround, comBySimbody);
-        
-    }
-}
 
-void testVelocityandAcceletationInGround()
-{
-    SimTK::Vec3 tolerance(SimTK::Eps);
-    
-    cout << "Running testVelocityandAcceletationInGround" << endl;
+    // Initialize the the sytem
+    SimTK::State state = pendulum->initSystem();
+
+    // Set the states randomly
+    SimTK::Random::Uniform random(-1, 1);
```

Comment:
I suggest hard-coding random numbers rather than picking them randomly each time a test is run. In theory, choosing test values randomly each time could cause occasional test failures that are difficult to track down.